lm_eval/models/nemo.py

```diff
@@ -0,0 +1,418 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import importlib
+import pathlib
+import filelock
+from typing import List, Literal
+from tqdm import tqdm
+import torch
+import numpy as np
+from copy import deepcopy
+
+from lm_eval.api.instance import Instance
+from lm_eval.api.model import LM
+from lm_eval.api.registry import register_model
+from lm_eval.models.utils import Collator
+from lm_eval.utils import simple_parse_args_string, get_rolling_token_windows, make_disjoint_window
+
+
+def _patch_pretrained_cfg(pretrained_cfg, trainer, tensor_model_parallel_size, pipeline_model_parallel_size):
+    import omegaconf
+    omegaconf.OmegaConf.set_struct(pretrained_cfg, True)
+    with omegaconf.open_dict(pretrained_cfg):
+        attributes_to_update = {
+            "sequence_parallel": False,
+            "activations_checkpoint_granularity": None,
+            "activations_checkpoint_method": None,
+            "precision": trainer.precision,
+            "global_batch_size": None,
+            "tensor_model_parallel_size": tensor_model_parallel_size,
+            "pipeline_model_parallel_size": pipeline_model_parallel_size,
+            "apply_rope_fusion": False
```

Comment:
Is there documentation / a comment we could add to code on why these attributes need be overridden (e.g., why is rope fusion not supported? is there somewhere an interested user could read up on this?)

Reply:
The reason of setting `"apply_rope_fusion": False` is simple: we are still testing it and fixing some bugs in nemo. Once it fully works, I'll delete this override.

The best place to read about this options is the config file https://github.com/NVIDIA/NeMo/blob/main/examples/nlp/language_modeling/conf/megatron_gpt_config.yaml.

Let me come back to you about `apply_rope_fusion` and also `sequence_parallel`, which is the other that we override.